### ISSUE TITLE
 Zero Trust tunnel supporting FTP?

### ISSUE THREAD

**Issue by lolTipsBR** [cloudflare/cloudflared#827]
I have a local NAS server and using the Tunnel to access the external internet.
I want to know if there is support for using FTP with cloudflared tunnel.

I tried looking in the documentation but couldn’t find it.

The ftp is working on the local network, without problems, using the local ip.
I want to access using the domain I configured in the tunnel.

The tunnel is already working, my domain is already pointed to a simple wordpress application.
I wanted support to use ftp with the domain I configured in the tunnel.

I've already made tcp settings pointing to localhost:21 , localhost:22 (ssh) and none of them work.

**Comment by morpig** [comment]
yes it's possible for ftp (any tcp basically), but you need to use cloudflared on the client side too.

**Comment by lolTipsBR** [comment]
> yes it's possible for ftp (any tcp basically), but you need to use cloudflared on the client side too.

Could you explain?
I have cloudflared installed on my NAS server. Do I need to install it on the computer that will access the NAS?
Do I need any more configuration? I couldn't find this information in the documentation.

Could you help me?

**Comment by fernvenue** [comment]
Hi @lolTipsBR, yes it does, actually `cloudflared` support any kind of TCP protocols, and I think you should read [the documentation](https://developers.cloudflare.com/cloudflare-one/applications/non-http/arbitrary-tcp/) first to get more information about that.

**Comment by APPX-Studio** [comment]
i use ftp clinet can't connect.
